enhancements/oc/must-gather-server-side-configuration.md

```diff
@@ -0,0 +1,397 @@
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2025-02-17
+last-updated: 2025-02-17
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/RFE-6505 # A generic RFE for now, this enhancement is a small step in bigger scheme
+see-also:
+  - "/enhancements/oc/must-gather.md"
+---
+
+To get started with this template:
+1. ~~**Pick a domain.** Find the appropriate domain to discuss your enhancement.~~
+1. ~~**Make a copy of this template.** Copy this template into the directory for
+   the domain.~~
+1. ~~**Fill out the metadata at the top.** The embedded YAML document is
+   checked by the linter.~~
+1. ~~**Fill out the "overview" sections.** This includes the Summary and
+   Motivation sections. These should be easy and explain why the community
+   should desire this enhancement.~~
+1. **Create a PR.** Assign it to folks with expertise in that domain to help
+   sponsor the process.
+1. **Merge after reaching consensus.** Merge when there is consensus
+   that the design is complete and all reviewer questions have been
+   answered so that work can begin.  Come back and update the document
+   if important details (API field names, workflow, etc.) change
+   during code review.
+1. **Keep all required headers.** If a section does not apply to an
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+See ../README.md for background behind these instructions.
+
+Start by filling out the header with the metadata for this enhancement.
+
+# Cluster side configuration for must-gather
+
+## Summary
+
+The idea is to allow configuration of must gather via an object deployed to
+a cluster. This allows better match to specific data collection needs based
+on the purpose of the cluster and avoids human mistakes when running must-gather
+manually.
+
+## Motivation
+
+Many customer cases and bug reports I have been exposed to lack the proper
+data needed for debugging the cluster at hand.
+
+1. Sometimes it is caused by humans not following the right procedure
+   (such as adding an extra must-gather image to the command line),
+1. in other cases it was caused by special needs that must gather by default
+   cannot satisfy for scale reasons (eg. low level kernel details needed to
+   debug Telco RAN latency on single node clusters).
```

Comment:
There is already an [`--all-images`](https://github.com/openshift/oc/blob/beeaf0904b6e9a7c22a73af9e92327f50a4d6918/pkg/cli/admin/mustgather/mustgather.go#L127) flag on the must-gather command. Isn't that flag be enough to accomplish this?

See proposal: https://github.com/openshift/enhancements/pull/1487